pkg/controller/controller_test.go

```diff
@@ -0,0 +1,102 @@
+package controller
```

Comment:
This test seems to test functions from `watchnamespaces.go` so I  am wondering whether it would be better to rename this file to `watchnamespaces_test.go` (instead of `controller_test.go`). WDYT?